ow_plexil/src/plexil-adapter/OwInterface.cpp

```diff
@@ -869,6 +885,66 @@ void OwInterface::guardedMoveAction (double x, double y, double z,
      guarded_move_done_cb<GuardedMoveResultConstPtr> (Op_GuardedMove));
 }
 
+void OwInterface::armMoveJoint (bool relative,
+                                int joint, double angle,
+                                int id)
+{
+  if (! markOperationRunning (Op_ArmMoveJoint, id)) return;
+  thread action_thread (&OwInterface::armMoveJointAction,
+                        this, relative, joint, angle, id);
+  action_thread.detach();
+}
+
+void OwInterface::armMoveJointAction (bool relative,
+                                      int joint, double angle,
+                                      int id)
+{
+  ArmMoveJointGoal goal;
+  goal.relative = relative;
+  goal.joint = joint;
+  goal.angle = angle;
+  
+  runAction<actionlib::SimpleActionClient<ArmMoveJointAction>,
+            ArmMoveJointGoal,
+            ArmMoveJointResultConstPtr,
+            ArmMoveJointFeedbackConstPtr>
+    (Op_ArmMoveJoint, m_armMoveJointClient, goal, id,
+     default_action_active_cb (Op_ArmMoveJoint),
+     default_action_feedback_cb<ArmMoveJointFeedbackConstPtr> (Op_ArmMoveJoint),
+     default_action_done_cb<ArmMoveJointResultConstPtr> (Op_ArmMoveJoint));
+} 
+
+void OwInterface::armMoveJoints (bool relative,
+                                 const vector<double>& angles,
+                                 int id)
+{
+  if (! markOperationRunning (Op_ArmMoveJoints, id)) return;
+  thread action_thread (&OwInterface::armMoveJointsAction,
+                        this, relative, angles, id);
+  action_thread.detach();
+}
+
+void OwInterface::armMoveJointsAction (bool relative,
+                                       const vector<double>& angles,
+                                       int id)
+{
+
+  ArmMoveJointsGoal goal;
+  goal.relative = relative;
+  for(int i = 0; i < goal.angles.size(); i++){
+    goal.angles[i] = angles[i];
+  }
```

Comment:
Continuing our email discussion, I'm guessing the problem somehow starts here, and that the array getting assigned to `goal.angles` (or not) is not the one coming in (`angles`) which looks correct.  I tried a few easy experiments, and did a quick web search on this topic, no luck.

Reply:
I did some experimentation and am committing a simple solution.  One reason the existing code didn't work is that goal.angles.size() is unspecified at first -- the ROS array has no defined size.  I tried doing a resize(6) but this didn't work for some reason.